runtime/krest/src/lib.rs

```diff
@@ -428,20 +436,104 @@ impl WeightToFeePolynomial for WeightToFee {
 
 pub struct DealWithFees;
 impl OnUnbalanced<NegativeImbalance> for DealWithFees {
-	fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalance>) {
-		if let Some(mut fees) = fees_then_tips.next() {
-			if let Some(tips) = fees_then_tips.next() {
-				tips.merge_into(&mut fees);
-			}
-			// pay fees to collators
-			<ToStakingPot as OnUnbalanced<_>>::on_unbalanced(fees);
-		}
+	// Overwrite on_unbalanced() and on_nonzero_unbalanced(), because their default
```

Comment:
I checked the default implementation of `OnUnbalanced` [here](https://github.com/peaqnetwork/substrate/blob/95067ccc3a7683768f8a10036785cde8d5f8a098/frame/support/src/traits/tokens/imbalance/on_unbalanced.rs#L32-L53).

By default the `on_unbalanceds` merges the iterator of imbalances and passes them to `on_unbalanced`, and this method may pass it to `on_nonzero_unbalanced` if the drop fails, and then this method drops the imbalance... Because we don't want to drop our imbalances, I prefer implementing the last two methods to avoid this properly...